cdc/log.cc

```diff
@@ -1463,22 +1463,25 @@ class transformer final : public change_processor {
 
         assert(_builder);
 
-        auto image_ck = _builder->allocate_new_log_row(op);
-        if (ck) {
-            _builder->set_clustering_columns(image_ck, *ck);
-        }
-
         const auto kind = ck ? column_kind::regular_column : column_kind::static_column;
 
         cell_map* row_state;
         if (ck) {
             row_state = get_row_state(_clustering_row_states, *ck);
-            if (!row_state) {
+            if (!row_state && op == cdc::operation::pre_image) {
```

Comment:
Why is this specific to pre_image? What about post_image? It can also be missing because the operation was to delete a row, in which case there is no post_image, no?
(the Alternator tests for that is missing, by the way, I still need to write them).

Reply:
We always generate post image, for better or for worse. Though technically it might be more reasonable to _not_ do this. Right now, this is preserving the behaviour as-is except for pre-image where I need it to change. I think follow-up...

Reply:
Ok, I'll accept this. If there is also a similar bug in the postimage, we'll discover it later when I write the test, and fix it later.

Reply:
Removed the check. Postimage is not consistent with pre-image, i.e. none if the row does not exist.